### DISCUSSION TITLE
Devices, Send/Sync

### DISCUSSION THREAD

**Discussion by pmathia0** [ash-rs/ash#500] [Q&A] [answered]
Hello,
could someone please explain difference between ash::Device and ash::vk::Device?

Also, why do they implement Send and Sync traits? The spec says the access to VkDevice must be externally synchronized. Am I missing something?

Thank you

**Comment by nickkuk** [comment, marked as answer]
`ash::vk::Device` is just 64-bit Vulkan handle; `ash::Device` is `ash::vk::Device` plus addresses of loaded functions.

In Vulkan spec chapter `Threading Behavior` you can see that in Vulkan only parameters of some functions must be externally synchronized. For `VkDevice` it is only `vkDestroyDevice` and while using `ash` you must do this synchronization yourself, e.g. just create `impl Drop` for your device wrapper (which also can implement `Send` and `Sync`).

**Reply by pmathia0** [reply]
I see.  Thank you for explanation!
